synbiohub_adapter/upload_sbol/upload_sbol.py

```diff
@@ -355,15 +386,18 @@ def __merge_remote_sub_collections(self, docs, collection_namespace):
                 remote_uri = '/'.join([collection_namespace, sub_collection.displayId, sub_collection.version])
 
                 try:
-                    self.part_shop.pull(remote_uri, remote_doc, False)
+                    if self.spoofed_url:
+                        self.part_shop.pull(remote_uri.replace(self.spoofed_url, self.url), remote_doc, False)
+                    else:
+                        self.part_shop.pull(remote_uri, remote_doc, False)
 
                     try:
-                        remote_sub_collection = remote_doc.getCollection(remote_uri.replace(self.url + '/', ''))
+                        remote_sub_collection = remote_doc.getCollection(remote_uri)
 
                         sub_collection.members = sub_collection.members + remote_sub_collection.members
-                    except:
+                    except RuntimeError:
```

Comment:
This is a very broad exception to catch. Is there a more narrow exception, specific to the situation at hand, that could be caught instead? Could the underlying error be passed along for context in the raised SubCollectionMergeError?

Reply:
Unfortunately, I don't think there is a more specific exception to catch. I will comment as best I can.

Reply:
Instead of commenting, adding message from RuntimeError to SubCollectionMergeError for additional context.